services/server/server.go

```diff
@@ -493,6 +514,67 @@ func (s *Server) ChangeSettings(ctx context.Context, req *serverpb.ChangeSetting
 	return res, nil
 }
 
+func (s *Server) validateSSHKey(sshKey string) error {
+	tempFile := path.Join(os.TempDir(), "temp_keys")
+	err := ioutil.WriteFile(tempFile, []byte(sshKey), os.FileMode(0600))
+	if err != nil {
+		return err
+	}
+
+	cmd := exec.Command("ssh-keygen", "-l", "-f", tempFile)
```

Comment:
We have to have a timeout there. The easiest ways seems to be to pass the request context with added timeout.